src/ch/blinkenlights/android/vanilla/MediaUtils.java

```diff
@@ -426,12 +428,36 @@ public static Song getRandomSong(Context context, boolean albumShuffle) {
 			sSongCount = songs.size();
 		}
 
-		Song result = null;
+		final List<Song> results = new ArrayList<>();
+
 		if (songs.size() > 0) {
-			result = songs.remove(0);
-			result.flags |= Song.FLAG_RANDOM;
+			final Song song = songs.remove(0);
+
+			if (!albumShuffle) {
+				song.flags |= Song.FLAG_RANDOM;
```

Comment:
The current logic is to remove auto-added songs if the user disables the random mode or adds new songs.
Guess it's okay to keep albums in the queue (even if they were auto added), but maybe add a comment about this.

Reply:
Done so. I could investigate in future popping in a new flag to say the song is from a random album and allow it to be removed if the album is not currently playing?